### DISCUSSION TITLE
Iterative method for thermal conductivity and four phonon scattering

### DISCUSSION THREAD

**Discussion by ZhunyunTang99** [phoebe-team/phoebe#189] [Q&A] [answered]
Dear jenny

I have been following Phoebe since it was first released, but I haven't used it because the features that can be implemented from the beginning are relatively simple. I focus on the thermal transport of materials, and I have always used ShengBTE or fourphonon code before, because it can solve the three phonon scattering through the Iterative method and the four phonon scattering through the RTA method to solve the thermal conductivity, while Phoebe does not seem to realize the four phonon solution. However, recently I have seen that Phoebe seems to have implemented the incorporation of electroacoustic coupling scattering into the thermal conductivity solving process, which is currently not a feature of ShengBTE. Here, I want to ask whether Phoebe can also solve three phonon scattering by Iterative method and four phonon scattering by RTA method at present. If these can be achieved, I believe Phoebe will be a very powerful software. Looking forward to your reply.

Best regards
Tang

**Comment by jcoulter12** [comment, marked as answer]
Hi Tang, 

Sorry for a slightly delayed reply to your message. 
Yes, right now we don't include 4-phonon scattering -- though it is something we'd like to do, we're a small development team, and we don't have a timeline for 4-phonon scattering at present. We appreciate that you've been following the package developments and that you're writing to let us know you'd like to see this feature, as this helps us keep track of what users want to see.  

By electroacoustic coupling, do you mean our recent addition of phonon-electron scattering to the thermal conductivity calculation? If that's what you mean, I would suspect there are almost no situations where you need both 4-phonon scattering and phonon-electron scattering simultaneously, as they apply in quite different temperature regimes. 

Additionally, I think it's worth noting that there's a few benefits to Phoebe -- when used correctly, it's quite computationally efficient. Also, it has some features, for example the relaxons solver and the simultaneous inclusion of ph-el and ph-ph scattering, which aren't available elsewhere and 

Thanks for writing, and feel free to follow up with further questions. You can register your interest in 4-phonon scattering on this [new git issue I opened as well](https://github.com/mir-group/phoebe/issues/190), by reacting with a thumbs up or commenting. 

Best, 
Jenny

**Reply by ZhunyunTang99** [reply]
Thank you for your reply.
Yes, the electroacoustic coupling I mentioned is the addition of electron phonon scattering to the solution of the lattice thermal conductivity. My current interest is to have a software that can include both three-phonon scattering, four-phonon scattering and electron-phonon scattering in the solution process.
In addition, I would also like to ask if Phoebe can use an iterative approach in solving the three-phonon scattering process similar to the ShengBTE software.

Best，
Tang

**Reply by jcoulter12** [reply]
Hi Tang, 

Yes, Phoebe can perform the iterative solution. We actually offer 5 levels of BTE solvers for both electron and phonon transport

1) Constant relaxation time approximation (not recommended)
2) RTA
3) [Simple iterative (Omini-Sparavigna) ](https://phoebe.readthedocs.io/en/develop/theory/phononBTE.html#iterative-solution-omini-sparavigna-method) -- convergence not guaranteed, but it's fast because it can use the full symmetries of the BTE if useSymmetries = true 
4) [Variational iterative](https://phoebe.readthedocs.io/en/develop/theory/phononBTE.html#iterative-solution-variational-method) -- based on conjugate gradient method, with more stable, fast convergence but unable to use the symmetries of the BTE. 
5) [Relaxons solver](https://phoebe.readthedocs.io/en/develop/theory/phononBTE.html#relaxons-solution) -- a direct diagonalization of the scattering matrix is used to solve the BTE, useful for the analysis of electron or phonon hydrodynamic systems. 

We also provide the [Wigner transport correction](https://phoebe.readthedocs.io/en/develop/theory/phononBTE.html#wigner-correction-to-phonon-thermal-conductivity) to the RTA by default in the output. 

Users can select these by setting this input variable: [solverBTE](https://phoebe.readthedocs.io/en/develop/inputFiles.html#solverbte)

For a [phonon transport input file,](https://phoebe.readthedocs.io/en/develop/inputFiles.html#phonon-bte-solver) in case you try out the code, I've attached a recommended starting point input file that I've been planning to upload to our tutorials. 

Let me know if you have further questions, 
Jenny

```

# triggers the phonon transport application 
appName = "phononTransport"

# set your 2nd and 3rd order force constant files 
phFC2FileName = "./ForceConstants2nd"
phFC3FileName = "./ForceConstants3rd"
sumRuleFC2 = "crystal"   # could choose "simple" ASR as well 

# basic BTE solution parameters
qMesh = [10,10,10]   # increase this to converge, of course
temperatures = [300.]
smearingMethod = "adaptiveGaussian"
#boundaryLength = 10. mum  # uncomment to set boundary scattering 

# BTE solver method
# RTA is always run regardless of solverBTE options 
solverBTE = ["iterative"]   
# if the scattering matrix should be stored in memory or not
# if you run out of memory, set it to false
scatteringMatrixInMemory = true
useSymmetries = true
# filter out states which occupation factors will make
# irrelevant to transport -- important for low-T calculations
windowType = “population”

# defaults that are also set automatically
# withIsotopeScattering = true
# dimensionality = 3
# convergenceThresholdBTE = 1.0e-5
# maxIterationsBTE = 50 
```

**Reply by ZhunyunTang99** [reply]
Thank you for your reply.

Finally, I look forward to the early realization of fourphonon scattering.



Good luck

Tang
